src/plots/cartesian/axes.js

```diff
@@ -3770,16 +3788,28 @@ axes.getPxPosition = function(gd, ax) {
     var side = ax.side;
     var anchorAxis;
 
+    // Shift in the opposite direction depending on which side the axis is on
+    // But don't shift if 'position' is specified
+    var xshift = ax.position > 0 ? 0 : ax._xshift;
+    xshift = side === 'left' ? xshift : -xshift;
+
     if(ax.anchor !== 'free') {
-        anchorAxis = ax._anchorAxis;
+        if(axLetter === 'y') {
+            anchorAxis = {
+                _offset: ax._anchorAxis._offset - xshift,
+                _length: ax._anchorAxis._length
+            };
```

Comment:
`ax._anchorAxis` contains several other keys and values that may be of interest for other functions.

Wondering it might be better to use `Lib.extendFlat` e.g.
```js
            anchorAxis = extendFlat({}, ax._anchorAxis, {
                _offset: ax._anchorAxis._offset - xshift,
                _length: ax._anchorAxis._length
            });
```
or alternatively mutate `ax._anchorAxis` directly i.e.
```js
            ax._anchorAxis._offset = ax._anchorAxis._offset - xshift;
            ax._anchorAxis._length = ax._anchorAxis._length;
```
?

Reply:
I ran into difficulties with the latter in that this function is called multiple times during the axis drawing so the offset value ends up as double what it should be. 

Could you clarify the issue with the modification here? I'm not totally understanding as this seems to be following the same pattern as was already implemented eg. in lines 3809-3817 below.